src/main/scala/viper/gobra/frontend/info/implementation/resolution/Enclosing.scala

```diff
@@ -30,6 +30,37 @@ trait Enclosing { this: TypeInfoImpl =>
     case _ => id
   })
 
+  lazy val enclosingLoop: PNode => Option[PForStmt] = {
+    down[Option[PForStmt]](None){ case x: PForStmt => Some(x) }
+  }
+
+  // Returns the enclosing loop that has a specific label
+  // It also returns the invariants of that loop
+  def enclosingLabeledLoop(label: PLabelUse, node: PNode) : Option[PForStmt] = {
+    enclosingLoop(node) match {
+      case None => None
+      case Some(encLoop) => encLoop match {
+        case tree.parent(l: PLabeledStmt) if l.label.name == label.name => Some(encLoop)
+        case _ => enclosingLabeledLoop(label, tree.parent(encLoop).head)
```

Comment:
What does `head` do here?
Is this equivalent to this?
```suggestion
        case tree.parent(p) => enclosingLabeledLoop(label, p)
```